### PR TITLE
refactor: Trying to improve adaptive linear solver tolerance selection

### PR DESCRIPTION
**Pull request by paveltomin**
1. Pull out hard-coded params to be able to modify them from input:
```
- strongestTol = 1e-8 (default value)
- adaptiveGamma = 0.1
- adaptiveExponent = 1.0
```
2. Adjust `altKrylovTol` logic. Not sure that new one is good but old one is really confusing.
3. Add output to see what tolerance is selected and why (triggered by `logLevel>0` in linear solver params).

### CODE REVIEW

**Comment by paveltomin** [comment]
I guess this PR is more a discussion point for now.
I found that adaptive tolerance approach is both quite useful for large scale realistic cases and at the same time the algorithm is quite confusing.
Trying to adjust it to make tolerance choice more understandable.
Please let me know what you think.

File: src/coreComponents/physicsSolvers/SolverBase.cpp [hunk 1]
```diff
@@ -672,22 +672,24 @@ bool SolverBase::lineSearchWithParabolicInterpolation( real64 const & time_n,
...
-  real64 const strongestTol = 1e-8;
-  real64 const exponent = 2.0;
-  real64 const gamma = 0.9;
-
-  real64 normRatio = newNewtonNorm / oldNewtonNorm;
-  if( normRatio > 1 )
-    normRatio = 1;
-
-  real64 newKrylovTol = gamma*std::pow( normRatio, exponent );
-  real64 altKrylovTol = gamma*std::pow( oldNewtonNorm, exponent );
```
Comment (paveltomin) [inline src/coreComponents/physicsSolvers/SolverBase.cpp:745]: this one is confusing, i replaced `oldNewtonNorm` but old linear solver tolerance, but not quite sure about that
`oldNewtonNorm` squared is usually way too tight

**Comment by victorapm** [comment]
Thanks, Pavel! Just checking... The algorithm is described [here](https://epubs.siam.org/doi/abs/10.1137/0917003?journalCode=sjoce3), correct?

**Comment by paveltomin** [comment]
> Thanks, Pavel! Just checking... The algorithm is described [here](https://epubs.siam.org/doi/abs/10.1137/0917003?journalCode=sjoce3), correct?

I think yes, but because I don't have access I can't say for sure.
I was reading this one https://softlib.rice.edu/pub/CRPC-TRs/reports/CRPC-TR94463.pdf which should be the same I guess.

**Comment by paveltomin** [comment]
any thoughts?

**Comment by victorapm** [comment]
I don't have much to add for now but I'll keep thinking about this. @oseikuffuor1 could you chime in?

**Comment by paveltomin** [comment]
ping

**Comment by paveltomin** [comment]
review anyone?
we don't use this feature that often but sometimes it's quite useful for bigger/longer cases

**Review by victorapm — APPROVED** [review]
Sorry! I thought I had approved this before

**Comment by rrsettgast** [comment]
@castelletto1 @CusiniM Can you guys review this?

**Comment by paveltomin** [comment]
rebaseline required only because of the new params, no real results change

**Review by rrsettgast — APPROVED** [review]

File: src/coreComponents/linearAlgebra/utilities/LinearSolverParameters.hpp [hunk 2]
```diff
@@ -124,6 +124,9 @@ struct LinearSolverParameters
 #endif
     integer useAdaptiveTol = false;   ///< Use Eisenstat-Walker adaptive tolerance
     real64 weakestTol = 1e-3;         ///< Weakest allowed tolerance when using adaptive method
+    real64 strongestTol = 1e-8;       ///< Strongest allowed tolerance when using adaptive method
+    real64 adaptiveGamma = 0.1;       ///< Gamma parameter for adaptive method
+    real64 adaptiveExponent = 1.0;    ///< Exponent parameter for adaptive method
```
Comment (castelletto1) [inline src/coreComponents/linearAlgebra/utilities/LinearSolverParameters.hpp:129]: The default values --  `exponent = 2.0` and `gamma = 0.9` -- have changed. I understand that the old exponent is probably too high. How did you choose the new values?

**Review by castelletto1 — APPROVED** [review]
Looks good to me, thank you!

File: src/coreComponents/linearAlgebra/utilities/LinearSolverParameters.hpp (hunk 2 above)
Comment (paveltomin) [reply src/coreComponents/linearAlgebra/utilities/LinearSolverParameters.hpp:129]: Yes, you are right.
I just found that exponent=2 is too aggressive but at the same time 0.9 is too weak.
Found 1.0 and 0.1 are more balanced but could be case dependent, so I pulled them out into the input.

**Pull request merged**